src/ExecutionPathTracer/ExecutionPathTracer.cs

```diff
@@ -55,39 +46,16 @@ public class ExecutionPathTracer
         /// </summary>
         public void OnOperationStartHandler(ICallable operation, IApplyData arguments)
         {
-            this.currentDepth++;
+            RuntimeMetadata? metadata = null;
 
-            // If `compositeTracer` is initialized, pass operations into it instead
-            if (this.compositeTracer != null)
+            var parentOp = (this.operations.Count > 0) ? this.operations.Peek() : null;
```

Comment:
You could make this a bit more compact and readable by directly using the condition that you're testing here:

```C#
var metadata = (this.operations.Count > 0) && !this.operations.Peek().IsMeasurement
    ? operation.GetRuntimeMetadata(arguments)
    : null;
```
That works because `&&` short-circuits, such that if `this.operations.Count == 0`, the `?:` already falls through.

Reply:
Thanks @cgranade! That's a really neat way of writing it :D